src/giftCards/GiftCardUpdate/GiftCardHistory/hooks/useGiftCardHistoryEvents.ts

```diff
@@ -1,13 +1,40 @@
+import { useUser } from "@dashboard/auth";
+import { GiftCardEventsQuery, PermissionEnum, useGiftCardEventsQuery } from "@dashboard/graphql";
 import { useContext } from "react";
 
 import { GiftCardDetailsContext } from "../../providers/GiftCardDetailsProvider";
 
-const useGiftCardHistoryEvents = () => {
+const APP_PERMISSIONS = [PermissionEnum.MANAGE_APPS];
+const USER_PERMISSIONS = [PermissionEnum.MANAGE_USERS, PermissionEnum.MANAGE_STAFF];
+
+interface GiftCardHistoryEvents {
+  id: string | undefined;
+  events: NonNullable<GiftCardEventsQuery["giftCard"]>["events"] | null | undefined;
+}
+
+const useGiftCardHistoryEvents = (): GiftCardHistoryEvents => {
+  const { user } = useUser();
+
+  const canSeeApp =
+    user?.userPermissions?.some(permission => APP_PERMISSIONS.includes(permission.code)) || false;
+  const canSeeUser =
+    user?.userPermissions?.some(permission => USER_PERMISSIONS.includes(permission.code)) || false;
```

Comment:
it can be a single function, that checks permission agains given array of permissions